### DISCUSSION TITLE
Edge attribute prediction

### DISCUSSION THREAD

**Discussion by jamilgafur** [pyg-team/pytorch_geometric#4960] [Q&A]
Morning, I am looking for resources on edge_attribute prediction; thus given a set of nodes and a connection matrix, determine the edges classification. I currently see a lot of work in predicting if an edge should exist, but not much on given an edge predict what it is classified as. Any code snippets or resources are greatly appreciated.

For similar discussions I see a lot on link prediction (exist or not), however not as much on link; I would like to know, given a known edge what does it classify as.

Thank you for your time

**Comment by rusty1s** [comment]
This can be generally tackled the same way: (1) You apply a GNN to learn new node representations, (2) use source and destination node representations to learn a label for the given link/pair of nodes:
```python
h = self.GNN(x, edge_index)
h_src = h[edge_index[0]]
h_dst = h[edge_index[1]]
edge_out = self.classifier(torch.cat([h_src, h_dst], dim=-1))
```

**Reply by jamilgafur** [reply]
I understand the last three lines, however I am not sure about getting h; 

given an input of nodes and their connections you can get their appropriate embeddings, however what would the embedding size be?

**Reply by rusty1s** [reply]
`h` refers to the node embeddings computed by the GNN, it is a `[num_nodes, num_features]` matrix. It holds the local subgraph information around each node.
